hibernate-core/src/main/java/org/hibernate/query/sqm/tree/domain/AbstractSqmPath.java

```diff
@@ -163,9 +177,11 @@ public SqmExpression<Class<? extends T>> type() {
 	@Override
 	@SuppressWarnings("unchecked")
 	public SqmPath<?> get(String attributeName) {
-
-		final SqmPathSource<?> subNavigable = getReferencedPathSource().getSubPathSource( attributeName );
-
+		final SqmPathSource<?> resolvedModel;
+		final SqmPathSource<?> pathSource = ( resolvedModel = getResolvedModel() ) != null ?
```

Comment:
When does `getResolvedModel()` return `null`? IMO it should never do that.

Reply:
It can happen for `SqmCteRoot` and  `SqmDerivedRoot` since `getModel()` returns null. An alternative would be overriding the `getResolvedModel` for those two but I was not sure what would be appropriate.

Also using `getReferencedPathSource()` instead of `getModel()` when the pathSource is not generic would work, but that would change the semantics of `getResolvedModel` for some cases IMO (e.g. treated paths).

PS: one example of how `@Nullable` would help :smile: